### DISCUSSION TITLE
Preload\populate COMMITS section when VSCode first launches

### DISCUSSION THREAD

**Discussion by mkanet** [gitkraken/vscode-gitlens#1274] [Feedback]
Pardon the newbie question, as I'm very new to Gitlens/GIT in VSCode.

When I first launch VSCode, I noticed briefly there's an clock icon for Sidebar's "Source Control".  However, it stops after a few seconds.  

Later on, when I click on Sidebar's "Source Control", I still have to wait for the COMMITS section to populate;; which seems to only happen _after_ I click on "Source Control".

Is it possible for the COMMITS section under "Source Control" to pre-emptively preload\populate immediately after I launch VSCode (without having to click on the "Source Control" icon)?  My Remote repository is over a very slow network (VPN).  It would be great to save some time by loading COMMITS in the background so it's already populated by the time I navigate to that section.

**Comment by eamodio** [comment]
Unfortunately, GitLens is by-design lazy loaded -- so you only pay for the features your use/see. This is also true of the VS Code APIs. While in theory GitLens could preload some of the Git specific data in anticipation, it would be quite a large amount of work, since it is against the current design. 

Are you using the VS Code remote features (e.g. SSH) to connect to your remote repository?

**Comment by mkanet** [comment]
@eamodio thank you for explaining and all your hard work maintaining this great VSCode extension.

I'm connecting to a remote Git repository via UNC (over VPN).  I wasn't even able to use this method effectively until just recently (when you fixed the below bug in a recent Insider version)  Thank you for this bug fix!!!

[Error retrieving line history from UNC path #1087](https://github.com/eamodio/vscode-gitlens/issues/1087)

**Reply by eamodio** [reply]
If possible, I would suggest trying out the Remote SSH, see [here](https://code.visualstudio.com/learn/develop-cloud/ssh-lab-machines) as in my experience it offers much better performance than UNC paths.

**Reply by mkanet** [reply]
Thank you for the suggestion.  Yes, I've tried Remote SSH.  It is fast.  However, in this case, the repository sits on a Windows file server at work; which I access via VPN.  Performance is not that bad though.
